### PR TITLE
Indexed joins range scans ignore non-equalities

### PR DESCRIPTION
**Pull request by max-hoffman**
When building index lookups, do not about when we see non-equality expressions. Instead, continue with the equality subset.

Ex:
```diff
 SELECT pk,i,f FROM one_pk RIGHT JOIN niltable ON pk=i and pk > 0
-	RightJoin((one_pk.pk = niltable.i) AND (one_pk.pk > 0))
-	 	├─ Table(one_pk)
-		 │   └─ columns: [pk]
-		 └─ Table(niltable)
-		     └─ columns: [i f]
+	Project(one_pk.pk, niltable.i, niltable.f)
+	     └─ RightIndexedJoin((one_pk.pk = niltable.i) AND (one_pk.pk > 0))
+		     ├─ Table(niltable)
+		     │   └─ columns: [i f]
+		     └─ IndexedTableAccess(one_pk)
+		         ├─ index: [one_pk.pk]
+		        └─ columns: [pk]
```

### CODE REVIEW

File: sql/analyzer/indexes.go [hunk 1]
```diff
@@ -249,7 +249,7 @@ func getIndexes(
 			// either return a merged index lookup for both values, or for neither. Returning either one leads to incorrect
 			// results.
 			if !canMergeIndexLookups(result, indexes) {
-				return nil, nil
+				continue
```
Comment (max-hoffman) [inline sql/analyzer/indexes.go:252]: @zachmu i think this fixes the static lookup case, will add more tests to verify the parent filter is always left untouched

File: enginetest/queries/query_plans.go [hunk 2]
```diff
@@ -553,19 +553,21 @@ var PlanTests = []QueryPlanTest{
 	},
 	{
 		Query: `SELECT * FROM MYTABLE JOIN OTHERTABLE ON i = i2 AND s > s2`,
-		ExpectedPlan: "InnerJoin((mytable.i = othertable.i2) AND (mytable.s > othertable.s2))\n" +
+		ExpectedPlan: "IndexedJoin((mytable.i = othertable.i2) AND (mytable.s > othertable.s2))\n" +
 			" ├─ Table(mytable)\n" +
 			" │   └─ columns: [i s]\n" +
-			" └─ Table(othertable)\n" +
+			" └─ IndexedTableAccess(othertable)\n" +
+			"     ├─ index: [othertable.i2]\n" +
```
Comment (zachmu) [inline enginetest/queries/query_plans.go:560]: When did we do this? I like it

File: sql/analyzer/indexed_joins.go [hunk 3]
```diff
@@ -763,21 +764,22 @@ func getJoinIndexes(
...
 		exprs := splitConjunction(jc.cond)
+		var eqs []sql.Expression
 		for _, expr := range exprs {
 			switch e := expr.(type) {
 			case *expression.Equals, *expression.NullSafeEquals, *expression.IsNull:
+				eqs = append(eqs, e)
 			case *expression.Not:
 				switch e.Child.(type) {
 				case *expression.Equals, *expression.NullSafeEquals, *expression.IsNull:
+					eqs = append(eqs, e)
```
Comment (zachmu) [inline sql/analyzer/indexed_joins.go:775]: This seems wrong. Write a test for <> or IS NOT NULL in the join condition

**Review by zachmu — APPROVED** [review]
LGTM, one suspicious thing

File: enginetest/queries/query_plans.go (hunk 2 above)
Comment (max-hoffman) [reply enginetest/queries/query_plans.go:560]: [i killed DecoratedNode one of the weeks you were out](https://github.com/dolthub/go-mysql-server/pull/1145)

File: sql/analyzer/indexes.go (hunk 1 above)
Comment (zachmu) [reply sql/analyzer/indexes.go:252]: Still think we need at test with <> as part of the actual join condition, e.g. `a.col <> b.col`

**Review by zachmu — APPROVED** [review]
Better, but still missing one specific test case

**Pull request merged**